src/arithmetic/algebraic_expression/utils.c

```diff
@@ -204,8 +204,63 @@ AlgebraicExpression *_AlgebraicExpression_GetOperand
 	return __AlgebraicExpression_GetOperand(root, operand_idx, &current_operand_idx);
 }
 
-void _AlgebraicExpression_FetchOperands(AlgebraicExpression *exp, const GraphContext *gc,
-										Graph *g) {
+static void _AlgebraicExpression_UseTransposedMatrices(AlgebraicExpression *root,
+													   const GraphContext *gc,
+													   Graph *g) {
+	assert(root);
+	Schema *s = NULL;
+	uint child_count = 0;
+	GrB_Matrix M = GrB_NULL;
+	const char *label = NULL;
+	AlgebraicExpression *operand = NULL;
+
+	if(root->type == AL_OPERATION) {
+		switch(root->operation.op) {
+		case AL_EXP_ADD:
+		case AL_EXP_MUL:
+			// Keep searching for a transpose operation.
+			child_count = AlgebraicExpression_ChildCount(root);
+			for(int i = 0; i < child_count; i++) {
+				_AlgebraicExpression_UseTransposedMatrices(CHILD_AT(root, i), gc, g);
+			}
+			break;
+
+		case AL_EXP_TRANSPOSE:
+			/* Get transposed matrix out of the graph object
+			 * if transposed operand is NULL used transposed adjacency matrix.
+			 * othewise get transposed relationship matrix if exists
+			 * otherwise use the zero matrix. */
+			assert(AlgebraicExpression_ChildCount(root) == 1);
+			operand = _AlgebraicExpression_OperationRemoveRightmostChild(root);
+			assert(operand->type == AL_OPERAND);
```

Comment:
This assertion will fail on valid addition operations, like:
`MATCH (a)-[e]->(b)<-[f:visited|friend]-(c) RETURN *`
We should be able to simply add transpose matrices as we would with standard edge matrices, I think.

Reply:
Did you managed to trigger this assertion?
the tree should be in a specific structure that is add nodes at the top, mul nodes in the middle and lastly transpose and operands.

Reply:
Yes, I triggered it with the query
`MATCH (a)-[e]->(b)<-[f:visited|friend]-(c) RETURN *`